frontend/ui/widget/inputdialog.lua

```diff
@@ -453,7 +456,8 @@ function InputDialog:init()
     end
 
     -- If we're fullscreen without a keyboard, make sure only the toggle button can show the keyboard...
-    if self.fullscreen and not self.keyboard_visible then
+    -- except for kindle, where we still want to activate it via shortcuts ("Shift" + "Home") when in need of symbols
+    if self.fullscreen and not self.keyboard_visible and not Device:hasSymKey() then
```

Comment:
Something about the behavior or the comment seems a bit suspicious. Shift+Home in this scenario is a "toggle button" so then shouldn't it simply send the `force_toggle` argument as `true`?

Reply:
where we still want to ~~activate~~ toggle it, is that better?

> shouldn't it simply send the force_toggle argument as true

I am not sure what you are referring with that bit.

Reply:
The implication seems to be that this "toggle button" thing is touch only. If so I see no reason any keyboard shouldn't be able to toggle it (not just hasSymKey) and presumably the same way that toggle button does it rather than with a hack.

But like I said, maybe it's the comment that's unclear.

Reply:
one thing I should say here is that when VK is hidden and in full screen (on kindle at least) the button with the arrow that should show the keyboard, _does not_ work, I am not sure if that is the case elsewhere so if anyone could check would be great (the shortcut does work with this though `and not Device:hasSymKey()`)

Reply:
@poire-z could you please go to "book specific style tweaks" and test the behaviour and see if there is anything you don't like.

btw, in an ideal world... the Menu key would open the 'CSS menu' but I'll let you worry about that another time (it would need to also work with the input field in focus...)

Reply:
Quick try on the emulator (just input text with the virtual keyboard and my PC keyboard, and taping with the mouse no buttons) feels it just behave as before.
What has changed ? What should I really look for ? (be honest if there's something I didn't see :))

Reply:
Well as stated earlier, when the vk is hidden (option “show virtual keyboard” not ticked), the button that is supposed to bring up the keyboard does not do it. I still prefer to have the keyboard shortcut available as it would not need to lose focus to bring it up (on kindle where symbols are not easily accessible) 

the shortcut works well however, if you press the button with the arrow, that messes up the shortcut which from that point forward, also won’t work. You need to completely close the style tweak screen and open it again.

Reply:
> the button that is supposed to bring up the keyboard does not do it

Indeed, so fix that :) (no idea)

>  I still prefer to have the keyboard shortcut available

Which is it ? Shift+Home doesn't work for me (where VK shown or not).

It also happens (before or after this PR) that when hiding VK, I can't input from my PC keyboard. When I bring it up again, PC keys work. May be related to SDL start/stopTextInput, @Frenzie ?

Reply:
> Which is it ? Shift+Home doesn't work for me (where VK shown or not).

that is expected, that is why we are commenting on this line `if self.fullscreen and not self.keyboard_visible and not Device:hasSymKey() then`

> Indeed, so fix that :) (no idea)

well I was hoping you would have an idea.

Reply:
As you were hoping, I had to find one :)
Looks like it is you that is prevented showing the keyboard when one taps on the button: both these logs show up when pressed (but only the first time it seems):
```lua
function InputDialog:onShowKeyboard(ignore_first_hold_release)
    -- NOTE: There's no VirtualKeyboard widget instantiated at all when readonly,
    --       and our input widget handles that itself, so we don't need any guards here.
    --       (In which case, isKeyboardVisible will return `nil`, same as if we had a VK instantiated but *never* shown).
    require("logger").warn("onwhowkeyboard")
    if Device:hasKeyboard() and G_reader_settings:isFalse("virtual_keyboard_enabled") then
    require("logger").warn("onwhowkeyboard but not enabled")
        return
    end
```
So, some logic to rethink about.

Reply:
I'll have my best people working on it ;)

Reply:
Sorry, I'm too busy at the moment :)

Reply:
> Looks like it is you that is prevented showing the keyboard when one taps on the button

tbf, that button has never really worked (on NT kindle at least). on K4 (not running this PR) it just sends focus back to the input box ...